flake8_quotes/__init__.py

```diff
@@ -158,42 +158,31 @@ def get_quotes_errors(self, file_contents):
             last_quote_char = token.string[-1]
             first_quote_index = token.string.index(last_quote_char)
             unprefixed_string = token.string[first_quote_index:]
+            is_multiquote_string = unprefixed_string[0] * 2 == unprefixed_string[1:3]
+            start_row, start_col = token.start
 
-            # If our string has a valid multiline start, then ignore it
-            #   (""")foo""" -> good (continue)
-            #   (''')foo' -> possibly bad
-            #   (")foo" -> possibly bad
-            #   (')foo' -> possibly bad
-            if unprefixed_string.startswith(self.config['good_multiline']):
-                continue
-
-            # If our string isn't a known bad multiline
-            #   (''')foo''' -> bad (see reporting at end)
-            #   (')foo' -> possibly bad
-            #   (")foo" -> possibly bad
-            if not unprefixed_string.startswith(self.config['bad_multiline']):
-                # If our string is a known good string, then ignore it
-                #   (')foo' -> good (continue)
-                #   (")foo" -> possibly bad
-                if unprefixed_string.startswith(self.config['good_single']):
+            if is_multiquote_string:
+                # If our string has a valid multiline start, then ignore it
+                if unprefixed_string.startswith(self.config['good_multiline']):
                     continue
 
-                # If our string is wrapping a known good string, then ignore it
-                #   "it(')s" -> good (continue)
-                #   "foo" -> possibly bad
+                yield {
+                    'message': 'Q001 Remove bad quotes from multiline string.',
+                    'line': start_row,
+                    'col': start_col,
+                }
+            else:
+                # This covers two cases:
+                # - the string starts and end with the good single quote e.g. 'asdf'
+                # - the string contains a good single quote e.g. "asdf's", which is also OK
```

Comment:
As a personal preference, I've moved back to the example-based comments. I feel they communicate points more effectively -- in past iterations on this project, the verbal comments only confuse me =/